### PR TITLE
Bug fix - Cannot select experience on HL when offline

### PR DESCRIPTION
**Pull request by gdamron**
The getter for `Task.IsComplete` throws an exception if `Task.IsFaulted` is `true` -- even if `Task.IsComplete` itself is true. Because of this, always check `Task.IsFaulted` first.

Note: In this fix, I make the assumption that if a task is cancelled, neither `onError` nor `onComplete` should be called.

### CODE REVIEW

File: Assets/Plugins/CreateAR/Commons/UwpHttpService.cs [hunk 1]
```diff
@@ -352,18 +352,18 @@ protected void ApplyJsonPayload(object payload, HttpRequestMessage request)
...
             {
-                onComplete(responseTask.Result);
+                onError(responseTask.Exception);
             }
-            else
+            else if (responseTask.IsCompleted)
             {
-                onError(responseTask.Exception);
+                onComplete(responseTask.Result);
             }
```
Comment (thegoldenmule) [inline Assets/Plugins/CreateAR/Commons/UwpHttpService.cs]: Isn't there a missing `else` here such that neither callback is called?

Comment (thegoldenmule) [reply Assets/Plugins/CreateAR/Commons/UwpHttpService.cs]: PS - I see your comment that maybe neither is called-- but is this ensured here? I am very nervous. Perhaps at least a comment explaining to future programmers that the else is not required.

Comment (gdamron) [reply Assets/Plugins/CreateAR/Commons/UwpHttpService.cs]: Looking at the Task docs a bit more, `IsComplete` will be true when `IsCancelled` is true. just using an else statement here would be equivalent. I guess the real question is whether cancellation should be considered success, failure, or something else. if it is its own state, is ignoring it the right thing to do? maybe with an explicit `yield break` for clarity.

Comment (thegoldenmule) [reply Assets/Plugins/CreateAR/Commons/UwpHttpService.cs]: How about passing an enum in the callback, and always calling a callback (even on abort)?

Comment (gdamron) [reply Assets/Plugins/CreateAR/Commons/UwpHttpService.cs]: 👍

**Pull request merged**